compiler/types.nim

```diff
@@ -695,6 +699,8 @@ proc lastOrd*(conf: ConfigRef; t: PType; fixedUnsigned = false): BiggestInt =
   of tyOrdinal:
     if t.len > 0: result = lastOrd(conf, lastSon(t))
     else: internalError(conf, "invalid kind for lastOrd(" & $t.kind & ')')
+  of tyUncheckedArray:
+    result = high(BiggestInt)
```

Comment:
From the user PoV nothing changes, the following snippet outputs the same results w/ and w/o this patch (this happens because of the implicit UA[char] -> CString conversion, for any other T there's no overloaded `high`/`low`/`len` so we can possibly treat this as an (hidden) implementation detail)

```nim
var x: ptr UncheckedArray[char]
echo len(x) # prints 0
echo high(x) # prints -1
```

From the compiler PoV this makes life simpler since we don't have to check for `tfUncheckedArray` every now and then and nobody has to deal with the `[0..-1]` ranges produced by the old `UncheckedArray` implementation.

Reply:
Ok.